src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/CsipRecord.kt

```diff
@@ -128,5 +134,67 @@ data class CsipRecord(
     this.referral = referral
   }
 
-  fun <T : DomainEvent<AdditionalInformation>> registerEntityEvent(entityEvent: BaseEntityEvent<T>) = apply { registerEvent(entityEvent) }
+  fun <T : DomainEvent<AdditionalInformation>> registerEntityEvent(entityEvent: BaseEntityEvent<T>) =
+    apply { registerEvent(entityEvent) }
+
+  fun registerCsipEvent(domainEvent: CsipEvent) = apply { registerEvent(domainEvent) }
+
+  fun create(
+    createCsipRecordRequest: CreateCsipRecordRequest,
+    csipRequestContext: CsipRequestContext,
+    incidentType: ReferenceData,
+    incidentLocation: ReferenceData,
+    referrerAreaOfWork: ReferenceData,
+    incidentInvolvement: ReferenceData,
+    contributoryFactors: Map<String, ReferenceData>,
+    reason: Reason = Reason.USER,
+    description: String = DomainEventType.CSIP_CREATED.description,
+  ): CsipRecord = apply {
+    referral = createCsipRecordRequest.toInitialReferralEntity(
+      this,
+      csipRequestContext,
+      incidentType,
+      incidentLocation,
+      referrerAreaOfWork,
+      incidentInvolvement,
+    ).apply {
+      createCsipRecordRequest.referral.contributoryFactors.forEach { factor ->
+        this.addContributoryFactor(
+          createRequest = factor,
+          factorType = contributoryFactors[factor.factorTypeCode]!!,
+          actionedAt = createdAt,
+          actionedBy = createdBy,
+          actionedByDisplayName = createdByDisplayName,
+          source = csipRequestContext.source,
+        )
+      }
+    }
+    addAuditEvent(
+      action = AuditEventAction.CREATED,
+      description = "CSIP record created via referral with ${referral!!.contributoryFactors().size} contributory factors",
+      actionedAt = createdAt,
+      actionedBy = createdBy,
+      actionedByCapturedName = createdByDisplayName,
+      source = csipRequestContext.source,
+      reason = reason,
+      activeCaseLoadId = csipRequestContext.activeCaseLoadId,
+      isRecordAffected = true,
+      isReferralAffected = referral != null,
+      isContributoryFactorAffected = referral!!.contributoryFactors().isNotEmpty(),
```

Comment:
Similarly, the model validation means there has to be at least one contributoryFactor so this can also be true. Now it's fair to say that the entity doesn't (and shouldn't) know about the model validation so we could add a `require(referral!!.contributoryFactors().isNotEmpty()` check.

The logic here isn't tested by the int tests so it will be classed as not covered by Codecov. If it uses the require check above, you could add a unit test for that logic.